src/pymorize/units.py

```diff
@@ -110,15 +98,24 @@ def handle_unit_conversion(da: xr.DataArray, rule: Rule) -> xr.DataArray:
     model_unit = rule.get("model_unit")
     from_unit = da.attrs.get("units")
     if model_unit is not None:
-        logger.debug(
+        logger.info(
             f"using user defined unit ({model_unit}) instead of ({from_unit}) from DataArray "
         )
         from_unit = model_unit
     handle_chemicals(from_unit)
     handle_chemicals(to_unit)
     new_da = da.pint.quantify(from_unit)
-    logger.debug(f"Converting units: {from_unit} -> {to_unit}")
-    new_da = new_da.pint.to(to_unit).pint.dequantify()
+    dimless = rule.get("dimensionless_unit_mappings", {})
+    cmor_variable = rule.get("cmor_variable")
+    if cmor_variable in dimless:
+        _to_unit = dimless[cmor_variable][to_unit]
```

Comment:
`to_unit` might not exists in the `dimensionless_unit_mappings` because we might have made a mistake while writing the yaml. Can you add an error handling here? Something like:

```python
if to_unit not int dimless[cmor_variable]:
    raise KeyError(f"Unit {to_unit} not found for variable {da.name} in the {pymorize_cfg.get("dimensionless_mapping_table", None)}")
```

I guess the pymorize_cfg get dimensionless_mapping_table won't work since it it out of scope, but it could be included in the rule, so that we can report in the error exactly in which file the unit is missing.

Reply:
good point. will write the error handling thing.